### DISCUSSION TITLE
Install issues for MacOS Monterey 12.6.3

### DISCUSSION THREAD

**Discussion by Kiguli** [AdaptiveCpp/AdaptiveCpp#1071] [Q&A] [answered]
Hello, I am trying to install Open SYCL for use on my Macbook. I understand it will only enable CPU parallelisation which isn't an issue.

I have installed LLVM on my machine and following the install steps I then get this error:

```
-- Configuring incomplete, errors occurred!
##@###### SYCLbuild % cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DLLVM_DIR=/usr/local/opt/llvm/cmake -DCMAKE_CXX_COMPILER=-fopenmp  ../OpenSYCL
-- Could not find HIP cmake integration, falling back to finding hipcc
-- Could not find ROCm installation by looking for hipcc. ROCm support unavailable.
CMake Error at /usr/local/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.26.4/share/cmake/Modules/FindOpenMP.cmake:577 (find_package_handle_standard_args)
  src/runtime/CMakeLists.txt:196 (find_package)
-- Configuring incomplete, errors occurred!
```

I assume it relates to my options attached to the CMAKE command. I would appreciate any help setting up the options correctly.
I believe I have correctly installed llvm, libomp, python3, clang, cmake and boost.

I appreciate any support that can be provided :)

**Comment by illuhad** [comment, marked as answer]
Hi,

I am not an expert for MacOS, but `-DCMAKE_CXX_COMPILER=-fopenmp` seems wrong to me, as this flag is used to set the C++ compiler that cmake uses for compilation, and `-fopenmp` is just a compiler flag, not the path to a C++ compiler.

You might also be missing `-DOpenMP_ROOT` which might be necessary on MacOS. I recommend checking out how we build it in our MacOS CI:
https://github.com/OpenSYCL/OpenSYCL/blob/develop/.github/workflows/macos.yml

**Reply by Kiguli** [reply]
Thanks that helped, I included the line from your link and it now works:

cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DLLVM_DIR=/usr/local/opt/llvm/cmake -DCMAKE_CXX_COMPILER=/usr/local/opt/llvm/bin/clang++ -DOpenMP_ROOT=`brew list libomp | grep libomp.a | sed -E "s/\/lib\/.*//"`  ../OpenSYCL
